### PR TITLE
[FLINK-21732][python] Bundle Java libraries into apache-flink-libraries

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will bundle Java libraries into `apache-flink-libraries`*

## Brief change log

  - *Add a new distributed package named `apache-flink-libraries` and bundle Java libraries into it*
  - *Change the pipeline of building `apache-flink` wheel packages in Azure Pipeline*

## Verifying this change

This change added tests and can be verified as follows:

  - *original IT tests and UT tests*
  - *Manual test installing `apache-flink-libraries` and `apache-flink` in a machine with old version of `apache-flink` and run some python udf test*
  - *trigger building `apache-flink` wheel packages in private Azure pipeline*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/setup.py [hunk 1]
```diff
@@ -43,15 +42,47 @@ def remove_if_exists(file_path):
...
+
+def extracted_output_files(base_dir, file_path, output_directory):
+    extracted_file_paths = []
+    from xml.dom.minidom import parse
+    dom = parse(file_path)
+    root_data = dom.documentElement
+    file_elements = (root_data.getElementsByTagName("files")[0]).getElementsByTagName("file")
+    # extracted <files><file></file></files>
+    for file_element in file_elements:
+        source = ((file_element.getElementsByTagName('source')[0]).childNodes[0]).data
```
Comment (WeiZhong94) [inline flink-python/setup.py:65]: Currently this function does not support the `<includes></includes>` tag and `<excludes></excludes>` tag. I think we need to raise Exception when the file element contains them to ensure the correctness.

**Review by WeiZhong94 — COMMENTED** [review]
@HuangXingBo Thanks for your PR. I have left some comments for you. In addition I think the content in `docs/flinkDev/building#build_pyflink` also need to be changed because it does not work on current situation.

**Comment by HuangXingBo** [comment]
@WeiZhong94 Thanks a lot for the review. I have addressed the comments at the latest commit.

**Review by WeiZhong94 — APPROVED** [review]
LGTM, +1 to Merge

**Comment by HuangXingBo** [comment]
https://dev.azure.com/hxbks2ks/FLINK-TEST/_build/results?buildId=982&view=results

**Pull request closed without merging**